lib/hooks.dart

```diff
@@ -396,3 +396,36 @@ T useContext<T>(Context<T> context) => ContextHelpers.unjsifyNewContext(React.us
 ///
 /// Learn more: <https://reactjs.org/docs/hooks-reference.html#useref>.
 Ref<T> useRef<T>([T initialValue]) => Ref.useRefInit(initialValue);
+
+/// Returns a memoized version of the return value of [createFunction].
+///
+/// If one of the [dependencies] has changed, [createFunction] is run during rendering of the [DartFunctionComponent].
+/// This optimization helps to avoid expensive calculations on every render.
+///
+/// > __Note:__ there are two [rules for using Hooks](https://reactjs.org/docs/hooks-rules.html):
+/// >
+/// > * Only call Hooks at the top level.
+/// > * Only call Hooks from inside a [DartFunctionComponent].
+///
+/// __Example__:
+/// ```
+/// UseMemoTestComponent(Map props) {
+///   final num = useState(0);
```

Comment:
#nit probably shouldn't use `num` as a variable name - since its a reserved word / primitive type in Dart's type system.